core/templates/dev/head/domain/StateAnswerStatsFactory.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for domain object which holds the statistics of a
+ * particular answer from some particular state.
+ */
+
+oppia.factory('StateAnswerStatsFactory', [function() {
```

Comment:
Conceptually there is no concept of a "state" here, and it's a single stat. Let's just call this an AnswerStat?

"StateAnswerStats" sounds to me like the full answer stats for all the states.

Reply:
How about StudentAnswerStats?

Reply:
Now it sounds like the stats for all answers submitted by a student :)

Reply:
Done.